src/liballoc/tests/vec.rs

```diff
@@ -1264,6 +1264,76 @@ fn test_try_reserve_exact() {
 
 }
 
+/// Tests that `recycle` successfully re-interprets the type to have different lifetime from the original
+#[test]
+fn test_recycle_lifetime() {
+    let s_1 = "foo".to_string();
+    let mut buf = Vec::with_capacity(100);
+    {
+        let mut buf2 = buf;
+        let s_2 = "foo".to_string();
+        buf2.push(s_2.as_str());
+
+        assert_eq!(buf2.len(), 1);
+        assert_eq!(buf2.capacity(), 100);
+
+        buf = buf2.recycle();
```

Comment:
What run-time properties is this actually checking other than successfully compiling?
I would expect the tests here to exercise size, alignment, truncation, etc. with assertions.

Reply:
Well it currently checks the length and capacity, which accounts for the size and truncation in your list. I'll add a check for alignment.

Reply:
More specifically I would expect a test that panics.

Reply:
Ah, I see. I have panicking tests after this; see `test_recycle_incompatible_size` and `test_recycle_incompatible_alignment`.